src/security-council-mgmt/SecurityCouncilUpgradeAction.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "./interfaces/IGnosisSafe.sol";
+import "./SecurityCouncilMgmtUtils.sol";
+
+/// @notice Action contract for updating security council members. Used by the security council management system.
+contract SecurityCouncilUpgradeAction {
+    address internal constant SENTINEL_OWNERS = address(0x1);
+
+    IGnosisSafe public immutable securityCouncil;
+
+    constructor(address _securityCouncil) {
+        securityCouncil = IGnosisSafe(_securityCouncil);
+    }
+
+    /// @notice updates members of security council multisig to match provided array
+    function updateMembers(address[] memory _updatedMembers) external {
+        // always preserve current threshold
+        uint256 threshold = securityCouncil.getThreshold();
+
+        address[] memory membersToAdd = new address[](6);
+        uint8 membersToAddCount = 0;
+
+        for (uint256 i = 0; i < _updatedMembers.length; i++) {
+            address member = _updatedMembers[i];
+            if (!securityCouncil.isOwner(member)) {
+                membersToAdd[membersToAddCount] = member;
+                membersToAddCount++;
+            }
+        }
+
+        address[] memory membersToRemove = new address[](6);
+        uint8 membersToRemoveCount = 0;
+
+        address[] memory owners = securityCouncil.getOwners();
+        for (uint256 i = 0; i < owners.length; i++) {
+            address owner = owners[i];
+            if (!SecurityCouncilMgmtUtils.isInArray(owner, _updatedMembers)) {
+                membersToRemove[membersToRemoveCount] = owner;
+                membersToRemoveCount++;
+            }
+        }
+
+        for (uint256 i = 0; i < membersToAddCount; i++) {
+            _addMember(membersToAdd[i], threshold);
+        }
+        for (uint256 i = 0; i < membersToRemoveCount; i++) {
+            _removeMember(membersToRemove[i], threshold);
+        }
+    }
+
+    function _addMember(address _member, uint256 _threshold) internal {
```

Comment:
I usually only use _ when I have conflict name. Its ok if we use this _ == internal style, but we havent done it much elsewhere. Happy to keep this though if you prefer it.

Reply:
mm okay yeah let's discuss and make consistent